ompi/mca/coll/tuned/coll_tuned_decision_fixed.c

```diff
@@ -72,25 +72,42 @@ ompi_coll_tuned_allreduce_intra_dec_fixed(const void *sbuf, void *rbuf, int coun
      *  {3, "recursive_doubling"},
      *  {4, "ring"},
      *  {5, "segmented_ring"},
-     *  {6, "rabenseifner"
+     *  {6, "rabenseifner"},
+     *  {7, "allgather_reduce"}
      *
      * Currently, ring, segmented ring, and rabenseifner do not support
      * non-commutative operations.
      */
     if( !ompi_op_is_commute(op) ) {
-        if (communicator_size < 4) {
-            if (total_dsize < 131072) {
-                alg = 3;
-            } else {
-                alg = 1;
-            }
-        } else if (communicator_size < 8) {
+        if (communicator_size == 2) {
```

Comment:
This logic only applies to disjoint communicators. FWICT communication patterns on the same node are faster with different algorithms.